3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/PiiPseudonymizerEnrichment.scala

```diff
@@ -334,25 +451,39 @@ final case class PiiJson(strategy: PiiStrategy,
   /**
    * Replaces a value in the given context data with the result of applying the strategy that value.
    */
-  private def jsonPathReplace(jValue: JValue): JValue = {
+  private def jsonPathReplace(jValue: JValue,
+                              strategy: PiiStrategy,
+                              schema: String): (JValue, List[JsonModifiedField]) = {
     val objectNode      = JsonMethods.mapper.valueToTree[ObjectNode](jValue)
     val documentContext = JJsonPath.using(JsonPathConf).parse(objectNode)
-    documentContext.map(
-      jsonPath,
-      ScrambleMapFunction(strategy)
-    )
-    JsonMethods.fromJsonNode(documentContext.json[JsonNode]())
+    val modifiedFields  = MutableList[JsonModifiedField]()
+    documentContext.map(jsonPath,
+                        ScrambleMapFunction(strategy, modifiedFields, fieldMutator.fieldName, jsonPath, schema))
+    (JsonMethods.fromJsonNode(documentContext.json[JsonNode]), modifiedFields.toList)
   }
 }
 
-case class ScrambleMapFunction(val strategy: PiiStrategy) extends MapFunction {
+case class ScrambleMapFunction(strategy: PiiStrategy,
+                               modifiedFields: MutableList[JsonModifiedField],
```

Comment:
I feel extremely bad about putting mutable data structures into `case class`es, especially non-private ones. I think this should `private class` at least.